### ISSUE TITLE
Setting fixed seed in PreprocessingLayer leads to different results 

### ISSUE THREAD

**Issue by zakajd** [tensorflow/tensorflow#49375]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version: 2.4.1
- Python version: 3.8
- CUDA/cuDNN version: Driver Version: 450.102.04   CUDA Version: 11.0
- GPU model and memory: V100

**Describe the current behavior**
I'm solving a task of image to image translation and would like to apply an identical augmentation to both the input and output images. For example, I'd like to flip both images, or rotate to the same degree. 
I can use transform from `tf.image.stateless_random_` and provide same seed value to get identical results, but there is no RandomTranslation or RandomRotation transforms. So I'm trying to use layers from `tf.keras.layers.experimental.preprocessing` with fixed seed.
But 2 layers initialised with same random seed are generating different transform (see example below). Is that expected?

**Describe the expected behavior**
Initialized from same seed 2 layers should do identical transformations.

**Standalone code to reproduce the issue**
```python
from tensorflow.keras import layers
import tensorflow as tf

transform_x = layers.experimental.preprocessing.RandomTranslation(height_factor=(-0.2, 0.3), width_factor=(-0.2, 0.3), seed=2)
transform_y = layers.experimental.preprocessing.RandomTranslation(height_factor=(-0.2, 0.3), width_factor=(-0.2, 0.3), seed=2)

def is_equal(x, y):
    return tf.reduce_sum(tf.cast(tf.math.equal(x, y), tf.float32)).numpy() == 0


x = tf.random.uniform((1, 32, 32, 3)) * 255.

for _ in range(15): 
    x_t = transform_x(x, training=True)
    y_t = transform_y(x, training=True)
    print(is_equal(x_t, y_t))
```

Running this code leads to the following result:
```
False
False
False
False
False
False
False
False
False
False
```

**Comment by zakajd** [comment]
Closing due to finding bug in my code. Everything works as expected.
